maskit/ensembles.py

```diff
@@ -0,0 +1,87 @@
+from enum import Enum
+from typing import Union, Dict
+
+from maskit.masks import PerturbationAxis, PerturbationMode, MaskedCircuit
+
+
+# TODO: support option "growing"
+class EnsembleMaskDefinitions(Enum):
+    RANDOM = {
+        "center": None,
+        "left": [
+            {"copy": {}},
+            {
+                "perturb": {
+                    "amount": 1,
+                    "mode": PerturbationMode.REMOVE,
+                    "axis": PerturbationAxis.RANDOM,
+                }
+            },
+        ],
+        "right": [
+            {"copy": {}},
+            {
+                "perturb": {
+                    "amount": None,
+                    "mode": PerturbationMode.INVERT,
+                    "axis": PerturbationAxis.RANDOM,
+                }
+            },
+        ],
+    }
+    CLASSICAL = {
+        "center": [
+            {"clear": {}},
+            {
+                "perturb": {
+                    "amount": 0.1,
+                    "mode": PerturbationMode.ADD,
+                    "axis": PerturbationAxis.RANDOM,
+                }
+            },
+        ],
+    }
+    EILEEN = {
+        "center": None,
+        "left": [
+            {"copy": {}},
+            {
+                "perturb": {
+                    "amount": 1,
+                    "mode": PerturbationMode.ADD,
+                    "axis": PerturbationAxis.RANDOM,
+                },
+            },
+        ],
+        "right": [
+            {"copy": {}},
+            {
+                "perturb": {
+                    "amount": 0.05,
+                    "mode": PerturbationMode.REMOVE,
+                    "axis": PerturbationAxis.RANDOM,
+                }
+            },
+        ],
+    }
+    GROWING = {"center": [{"shrink": {"amount": 1, "axis": PerturbationAxis.LAYERS}}]}
+
+
+def ensemble_branches(
+    dropout: Union[EnsembleMaskDefinitions, Dict],
+    masked_params: MaskedCircuit,
+):
+    branches = []
```

Comment:
Since you want to name the branches, consider to make this a dict. You can still access the ``values``, but can have names here.

Reply:
Thanks, I just changed the `list`s to `dict`s :)